object/virtual_machine.go

```diff
@@ -469,6 +469,41 @@ func (v VirtualMachine) RemoveDevice(ctx context.Context, keepFiles bool, device
 	return v.configureDevice(ctx, types.VirtualDeviceConfigSpecOperationRemove, fop, device...)
 }
 
+// AttachDisk attaches the given disk to the VirtualMachine
+func (v VirtualMachine) AttachDisk(ctx context.Context, id string, datastore *Datastore, controllerKey int32, unitNumber int32) error {
```

Comment:
I think the `unitNumber` param should also be `* int32`, otherwise it isn't possible to omit this optional field in the request. What do you think?

Reply:
I copied this from [vslm](https://github.com/vmware/govmomi/blob/f2eef8fc745fec4f654bbc85ea1c75bb88c24779/vslm/global_object_manager.go#L660) 
```
func (this *GlobalObjectManager) AttachDisk(ct context.Context, id vim.ID, vm mo.Reference, controllerKey int32,
	unitNumber int32) (*Task, error) {
```
I am ok either way though.

Reply:
ok, let's keep it this way for consistency.